skins/oasis/modules/CommentsLikesController.class.php

```diff
@@ -18,52 +19,48 @@ public function init() {
 	 * Are article comments enabled for context title?
 	 */
 	private function checkArticleComments() {
-		$this->isArticleComments = class_exists('ArticleComment') && ArticleCommentInit::ArticleCommentCheckTitle($this->contextTitle);
+		$this->isArticleComments = class_exists( 'ArticleComment' ) && ArticleCommentInit::ArticleCommentCheckTitle( $this->contextTitle );
 		return $this->isArticleComments;
 	}
 
 	/**
 	 * Get URL of the page comments button should be linking to
 	 */
 	private function getCommentsLink() {
-		wfProfileIn(__METHOD__);
-		global $wgTitle, $wgRequest;
+		wfProfileIn( __METHOD__ );
 
-		$isHistory = $wgRequest->getVal('action') == 'history';
+		$isHistory = $this->wg->Request->getVal( 'action' ) == 'history';
 
-		if ($this->checkArticleComments()) {
+		if ( $this->checkArticleComments() ) {
 			// link to article comments section
-			if ($this->contextTitle != $wgTitle || $isHistory) {
-				$commentsLink = $this->contextTitle->getLocalUrl() . '#WikiaArticleComments';
-			}
-			else {
+			if ( $this->contextTitle != $this->wg->Title || $isHistory ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)